src/Integration.Vsix/CFamily/VcxProject/VcxRequestFactory.cs

```diff
@@ -75,36 +75,28 @@ internal class VcxRequestFactory : IRequestFactory
 
         public async Task<IRequest> TryCreateAsync(string analyzedFilePath, CFamilyAnalyzerOptions analyzerOptions)
         {
-            IRequest request = null;
-            await threadHandling.RunOnUIThread(() => request = TryCreateSync(analyzedFilePath, analyzerOptions));
-            return request;
-        }
-
-        private IRequest TryCreateSync(string analyzedFilePath, CFamilyAnalyzerOptions analyzerOptions)
-        {
-            LogDebug("Trying to create request for " + analyzedFilePath);
+            threadHandling.ThrowIfOnUIThread();
```

Comment:
I'd forgotten we'd added this thread-handling abstraction. Handy.

Reply:
Definitely!